### PR TITLE
meson: install compat symlinks for systemctl

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

**Comment by yuwata** [comment]
lgtm.

File: meson.build [hunk 1]
```diff
@@ -1712,6 +1713,12 @@ exe = executable('systemctl', 'src/systemctl/systemctl.c',
                  install_dir : rootbindir)
 public_programs += [exe]
 
+foreach alias : ['halt', 'init', 'poweroff', 'reboot', 'runlevel', 'shutdown', 'telinit']
+        meson.add_install_script(meson_make_symlink,
+                                 join_paths(rootbindir, 'systemctl'),
```
Comment (yuwata) [inline meson.build]: Note. Currently, `init` is a symlink to `/usr/lib/systemd/systemd` on Fedora, not `systemctl`.

Comment (keszybz) [reply meson.build]: Oh, oh, oh. Thanks for catching this. Of course.

**Comment by keszybz** [comment]
Updated.

**Comment by yuwata** [comment]
lgtm.

BTW, I thought that `init` can be symlink to `systemctl`. Is it not true?

**Comment by keszybz** [comment]
No. /sbin/init must be a binary that can start the system, and systemctl has no code for that.

With sysvinit, /sbin/init also provided an implementation of /sbin/telinit, so /sbin/telinit was a symlink to /sbin/init. But we split those two roles, into systemd-init group, and systemctl-telinit-runlevel-... group.

**Comment by yuwata** [comment]
I see. Thank you for the explanation. Maybe an origin of my confusion is that `systemctl` accepts `init` instead of `telinit` as `program_invocation_short_name`. Please see #8305.

**Comment by poettering** [comment]
this breaks the rawhide CI, but I figure that's kinda expected.

Thanks for fixing this!

**Comment by bl33pbl0p** [comment]
It seems after this change the symlinks are now set up in /usr/sbin/ instead, and since that is a symlink to /usr/bin on Exherbo and Arch atleast, installation of the produced image/tarball fails (since it tries to write to /usr/sbin making it a directory while it should be a symlink).

**Comment by keszybz** [comment]
Oh, I didn't consider this case. Fedora is still suffering from the bin-sbin split.

**Comment by keszybz** [comment]
This is not a problem when installing into the real root, where /usr/sbin and /usr/bin already exist, right? It's only an issue when using DESTDIR?

**Comment by bl33pbl0p** [comment]
I think so, yes.

**Comment by keszybz** [comment]
https://github.com/systemd/systemd/pull/8319

**Comment by bl33pbl0p** [comment]
I'll try testing it.

**Comment by poettering** [comment]
@bl33pbl0p i merged #8319 now. BTW, if you merge bin+sbin like this (which I very much sympathize with), then i figure you might want to #ifdef out a few places in the systemd tree where we refer to the directory. I'd welcome a patch adding the necessary iffdeffery for dealing with that the same way as we already have `#ifdef HAVE_SPLIT_USR` at various places.

For example, /sbin is currently mentioned in src/core/namespace.c where it should suppressed if the dir is merged like that. Similar in src/basic/path-util.h and possibly others.

**Comment by keszybz** [comment]
https://github.com/systemd/systemd/pull/8324

**Pull request merged**